app/src/main/java/com/framgia/soundcloud/service/MusicService.java

```diff
@@ -0,0 +1,227 @@
+        mPlayer.setOnErrorListener(this);
+    }
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        return mIBinder;
+    }
+
+    @Override
+    public boolean onUnbind(Intent intent) {
+        mPlayer.stop();
+        mPlayer.release();
+        return false;
+    }
+
+    @Override
+    public void onCompletion(MediaPlayer mp) {
+        if (mp.isLooping()) {
+            return;
+        }
+        if (mp.getCurrentPosition() > 0) {
+            mp.reset();
+            playNext();
+        }
+    }
+
+    @Override
+    public boolean onError(MediaPlayer mp, int what, int extra) {
+        mp.reset();
+        return false;
+    }
+
+    @Override
+    public void onPrepared(MediaPlayer mp) {
+        mp.start();
+    }
+
+    @Override
+    public void onDestroy() {
+    }
+
+    public void playTrack() {
+        mPlayer.reset();
+        try {
+            mPlayer.setDataSource(mTracks.get(mPostion).getFullUri());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        mPlayer.prepareAsync();
+        updateSeekBar();
+        mMediaState = StateMode.STATE_PLAYING;
+    }
+
+    public void pauseTrack() {
+        if (!mPlayer.isPlaying()) return;
+        mPlayer.pause();
+        mResumePosition = mPlayer.getCurrentPosition();
+        mSeekBarHandler.removeCallbacks(mUpdateTime);
+        mMediaState = StateMode.STATE_PAUSE;
+    }
+
+    public void resumeTrack() {
+        if (mPlayer.isPlaying()) return;
+        mPlayer.seekTo(mResumePosition);
+        mPlayer.start();
+        updateSeekBar();
+        mMediaState = StateMode.STATE_PLAYING;
+    }
+
+    public void playPrev() {
+        if (mLoop == LoopMode.LOOP_TRACK) {
+            return;
+        }
+        mPostion--;
+        if (mPostion < 0) {
+            mPostion = 0;
+            if (mLoop == LoopMode.LOOP_LIST_TRACKS) mPostion = mTracks.size() - 1;
+        }
+        playTrack();
+    }
+
+    public void playNext() {
+        if (mLoop == LoopMode.LOOP_TRACK) {
+            return;
+        }
+        mPostion++;
+        if (mPostion >= mTracks.size()) {
+            mPostion = 0;
+        }
+        if (mIsSuffler) {
+            mPostion = mRandom.nextInt(mTracks.size() - 1);
+        }
+        playTrack();
+    }
+
+    public void setPostion(int postion) {
+        mPostion = postion;
+    }
+
+    public List<Track> getTracks() {
+        return mTracks;
+    }
+
+    public void setTracks(List<Track> tracks) {
+        mTracks = tracks;
+    }
+
+    public Track getCurrentTrack() {
+        return mTracks.get(mPostion);
+    }
+
+    @StateMode
+    public int getMediaState() {
+        return mMediaState;
+    }
+
+    public boolean isPlaying() {
+        return mPlayer.isPlaying();
+    }
+
+    @Retention(SOURCE)
+    @IntDef({ LoopMode.NONE_LOOP, LoopMode.LOOP_LIST_TRACKS, LoopMode.LOOP_TRACK })
+    public @interface LoopMode {
+        int NONE_LOOP = 0;
```

Comment:
ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.